src/controllers/index.js

```diff
@@ -1,15 +1,18 @@
 const express = require('express');
 
 const login = require('./login');
+const error = require('./error');
 const adminHomePage = require('./adminHomePage');
 const { authCheck } = require('./middleware');
 
 const router = express.Router();
 router.get('/admin/login', login.get);
 router.post('/admin/login', login.post);
 
-router.use(authCheck);
```

Comment:
Why did you stop authenticating for all routes and started using it only for `/admin` and `/admin/logout`? You were going in the right direction...

Reply:
because when go a route that doesn't exist it goes to the admin login (problem), so we check for auth in every excited route instead. and any other  not exited route will go to 404

Reply:
@ali-7 You should not be handling authentication for every route separately, you should keep it as you did it the first time, but rather than having it as a middleware for each route, you should have something like this:
```js
...
const openPages = ['/admin', '/admin/logout'];
router.use(openPages, authCheck);
...
```